src/jit/morph.cpp

```diff
@@ -3472,7 +3513,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
             regNumber nextOtherRegNum = REG_STK;
 
-            if (isStructArg)
```

Comment:
The predicate isStructArg should be sufficient here and always correct.
The proper fix is to make sure that isStructArg is true here, the second time we call this method..